### PR TITLE
upgrade logback from 1.0.7 to 1.3.14 and slf4j to 2.0.12

### PR DESCRIPTION
**Pull request by baulea**
# upgrade logback from 1.0.7 to 1.3.14

PR Branch
https://github.com/baulea/zerocode/tree/upgrade_logback

## Motivation and Context

- upgrade multiple maven dependencies for later Java17 compatibility, Issue https://github.com/authorjapps/zerocode/issues/607
- logback 1.0.7 has vulnerabilities, which are removed in 1.3.14: https://mvnrepository.com/artifact/ch.qos.logback/logback-classic
This also solves https://github.com/authorjapps/zerocode/pull/533 and https://github.com/authorjapps/zerocode/pull/499

## Checklist:

* [ ] Unit tests added

* [ ] Integration tests added

* [x] Test names are meaningful

* [x] Feature manually tested

* [x] Branch build passed

* [x] No 'package.*' in the imports

* [ ] Relevant Wiki page updated with clear instruction for the end user
  * [x] Not applicable. This was only a refactor change, no functional or behaviour changes were introduced

* [ ] Http test added to `http-testing` module(if applicable) ?
  * [x] Not applicable. The changes did not affect HTTP automation flow

* [ ] Kafka test added to `kafka-testing` module(if applicable) ?
  * [x] Not applicable. The changes did not affect Kafka automation flow

### CODE REVIEW

**Comment by a1shadows** [comment]
Capturing the diff in logback logs from another CI build and this one:

Older builds:
```[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running org.jsmart.zerocode.filebody.ReuseFileBodyTest
16:24:45,707 |-INFO in ch.qos.logback.classic.LoggerContext[default] - Could NOT find resource [logback.groovy]
16:24:45,707 |-INFO in ch.qos.logback.classic.LoggerContext[default] - Could NOT find resource [logback-test.xml]
16:24:45,708 |-INFO in ch.qos.logback.classic.LoggerContext[default] - Found resource [logback.xml] at [file:/home/runner/work/zerocode/zerocode/core/target/classes/logback.xml]
16:24:45,708 |-WARN in ch.qos.logback.classic.LoggerContext[default] - Resource [logback.xml] occurs multiple times on the classpath.
16:24:45,708 |-WARN in ch.qos.logback.classic.LoggerContext[default] - Resource [logback.xml] occurs at [file:/home/runner/work/zerocode/zerocode/core/target/classes/logback.xml]
16:24:45,708 |-WARN in ch.qos.logback.classic.LoggerContext[default] - Resource [logback.xml] occurs at [jar:file:/home/runner/.m2/repository/org/jsmart/micro-simulator/1.1.10/micro-simulator-1.1.10.jar!/logback.xml]
16:24:45,741 |-INFO in ch.qos.logback.classic.joran.action.ConfigurationAction - debug attribute not set
16:24:45,744 |-INFO in ch.qos.logback.core.joran.action.AppenderAction - About to instantiate appender of type [ch.qos.logback.core.FileAppender]
16:24:45,748 |-INFO in ch.qos.logback.core.joran.action.AppenderAction - Naming appender as [fileAppender]
16:24:45,764 |-INFO in ch.qos.logback.core.joran.action.NestedComplexPropertyIA - Assuming default type [ch.qos.logback.classic.encoder.PatternLayoutEncoder] for [encoder] property
16:24:45,794 |-INFO in ch.qos.logback.core.FileAppender[fileAppender] - File property is set to [target/logs/zerocode_rest_bdd_logs.log]
16:24:45,795 |-INFO in ch.qos.logback.core.joran.action.AppenderAction - About to instantiate appender of type [ch.qos.logback.core.ConsoleAppender]
16:24:45,796 |-INFO in ch.qos.logback.core.joran.action.AppenderAction - Naming appender as [consoleAppender]
16:24:45,801 |-WARN in ch.qos.logback.core.ConsoleAppender[consoleAppender] - This appender no longer admits a layout as a sub-component, set an encoder instead.
16:24:45,801 |-WARN in ch.qos.logback.core.ConsoleAppender[consoleAppender] - To ensure compatibility, wrapping your layout in LayoutWrappingEncoder.
16:24:45,801 |-WARN in ch.qos.logback.core.ConsoleAppender[consoleAppender] - See also http://logback.qos.ch/codes.html#layoutInsteadOfEncoder for details
16:24:45,801 |-INFO in ch.qos.logback.classic.joran.action.RootLoggerAction - Setting level of ROOT logger to INFO
16:24:45,801 |-INFO in ch.qos.logback.core.joran.action.AppenderRefAction - Attaching appender named [fileAppender] to Logger[ROOT]
16:24:45,801 |-INFO in ch.qos.logback.core.joran.action.AppenderRefAction - Attaching appender named [consoleAppender] to Logger[ROOT]
16:24:45,801 |-INFO in ch.qos.logback.classic.joran.action.ConfigurationAction - End of configuration.
16:24:45,802 |-INFO in ch.qos.logback.classic.joran.JoranConfigurator@481a15ff - Registering current configuration as safe fallback point
```

This build:

```
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running org.jsmart.zerocode.filebody.ReuseFileBodyTest
02:13:25,752 |-INFO in ch.qos.logback.classic.LoggerContext[default] - This is logback-classic version 1.3.14
02:13:25,753 |-INFO in ch.qos.logback.classic.util.ContextInitializer@501edcf1 - No custom configurators were discovered as a service.
02:13:25,753 |-INFO in ch.qos.logback.classic.util.ContextInitializer@501edcf1 - Trying to configure with ch.qos.logback.classic.joran.SerializedModelConfigurator
02:13:25,754 |-INFO in ch.qos.logback.classic.util.ContextInitializer@501edcf1 - Constructed configurator of type class ch.qos.logback.classic.joran.SerializedModelConfigurator
02:13:25,754 |-INFO in ch.qos.logback.classic.LoggerContext[default] - Could NOT find resource [logback-test.scmo]
02:13:25,754 |-INFO in ch.qos.logback.classic.LoggerContext[default] - Could NOT find resource [logback.scmo]
02:13:25,755 |-INFO in ch.qos.logback.classic.util.ContextInitializer@501edcf1 - ch.qos.logback.classic.joran.SerializedModelConfigurator.configure() call lasted 1 milliseconds. ExecutionStatus=INVOKE_NEXT_IF_ANY
02:13:25,755 |-INFO in ch.qos.logback.classic.util.ContextInitializer@501edcf1 - Trying to configure with ch.qos.logback.classic.util.DefaultJoranConfigurator
02:13:25,755 |-INFO in ch.qos.logback.classic.util.ContextInitializer@501edcf1 - Constructed configurator of type class ch.qos.logback.classic.util.DefaultJoranConfigurator
02:13:25,755 |-INFO in ch.qos.logback.classic.LoggerContext[default] - Could NOT find resource [logback-test.xml]
02:13:25,756 |-INFO in ch.qos.logback.classic.LoggerContext[default] - Found resource [logback.xml] at [file:/home/runner/work/zerocode/zerocode/core/target/classes/logback.xml]
02:13:25,756 |-WARN in ch.qos.logback.classic.util.DefaultJoranConfigurator@78b729e6 - Resource [logback.xml] occurs multiple times on the classpath.
02:13:25,756 |-WARN in ch.qos.logback.classic.util.DefaultJoranConfigurator@78b729e6 - Resource [logback.xml] occurs at [file:/home/runner/work/zerocode/zerocode/core/target/classes/logback.xml]
02:13:25,756 |-WARN in ch.qos.logback.classic.util.DefaultJoranConfigurator@78b729e6 - Resource [logback.xml] occurs at [jar:file:/home/runner/.m2/repository/org/jsmart/micro-simulator/1.1.10/micro-simulator-1.1.10.jar!/logback.xml]
02:13:25,835 |-INFO in ch.qos.logback.core.model.processor.AppenderModelHandler - Processing appender named [fileAppender]
02:13:25,835 |-INFO in ch.qos.logback.core.model.processor.AppenderModelHandler - About to instantiate appender of type [ch.qos.logback.core.FileAppender]
02:13:25,842 |-INFO in ch.qos.logback.core.model.processor.ImplicitModelHandler - Assuming default type [ch.qos.logback.classic.encoder.PatternLayoutEncoder] for [encoder] property
02:13:25,871 |-INFO in ch.qos.logback.core.FileAppender[fileAppender] - File property is set to [target/logs/zerocode_rest_bdd_logs.log]
02:13:25,872 |-INFO in ch.qos.logback.core.model.processor.AppenderModelHandler - Processing appender named [consoleAppender]
02:13:25,872 |-INFO in ch.qos.logback.core.model.processor.AppenderModelHandler - About to instantiate appender of type [ch.qos.logback.core.ConsoleAppender]
02:13:25,873 |-INFO in ch.qos.logback.core.model.processor.ImplicitModelHandler - Assuming default type [ch.qos.logback.classic.encoder.PatternLayoutEncoder] for [encoder] property
02:13:25,874 |-INFO in ch.qos.logback.classic.model.processor.RootLoggerModelHandler - Setting level of ROOT logger to INFO
02:13:25,874 |-INFO in ch.qos.logback.core.model.processor.AppenderRefModelHandler - Attaching appender named [fileAppender] to Logger[ROOT]
02:13:25,874 |-INFO in ch.qos.logback.core.model.processor.AppenderRefModelHandler - Attaching appender named [consoleAppender] to Logger[ROOT]
02:13:25,874 |-INFO in ch.qos.logback.core.model.processor.DefaultProcessor@6b4a4e18 - End of configuration.
02:13:25,875 |-INFO in ch.qos.logback.classic.joran.JoranConfigurator@27c86f2d - Registering current configuration as safe fallback point
02:13:25,875 |-INFO in ch.qos.logback.classic.util.ContextInitializer@501edcf1 - ch.qos.logback.classic.util.DefaultJoranConfigurator.configure() call lasted 120 milliseconds. 
```

**Comment by baulea** [comment]
Hi @a1shadows, the log entries logback outputs while reading its configuration files changed with the new version of logback. But the additional entries are only of type "INFO"l. Some of them are about a new feature called SerializedModelConfigurator like "Could NOT find resource [logback.scmo]", (https://logback.qos.ch/manual/configuration.html). Other log entries from the application itself did not change.
I could not find any problem with the new version of logback. @a1shadows , did you notice any problems?

**Comment by a1shadows** [comment]
> I could not find any problem with the new version of logback. @a1shadows , did you notice any problems?

No, looks good to me. Just thought to capture the log diff for future reference. Looks good to merge

File: junit5-testing/pom.xml [hunk 1]
```diff
@@ -12,11 +12,6 @@
     <name>Zerocode JUnit5 Jupiter Load Testing</name>
     <description>Zerocode tests with JUnit5 Jupiter Engine</description>
 
-    <properties>
```
[the remark this answers is not in the document]
Comment (a1shadows) [reply junit5-testing/pom.xml:15]: I'm guessing these are coming from the parent pom, hence we are removing them here

**Review by a1shadows — APPROVED** [review]
LGTM

**Pull request merged**